File: 1528. Shuffle String.py
#* 1528. Shuffle String
s = "codeleet"; indices = [4,5,6,7,0,2,1,3]
# @param {String} s
# @param {Integer[]} indices
# @return {String}
#* SOL 1
class Solution:
    def restoreString(self, s, indices):
        return ''.join(s[indices.index(i)] for i in range(len(indices)))
    
print(Solution().restoreString(s, indices))  # Output: "leetcode"

File: 1528. Shuffle String.js
// 1528. Shuffle String
s = "codeleet"; indices = [4,5,6,7,0,2,1,3]
// SOL 2
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
  return indices.map((index, i) => s[indices.indexOf(i)]).join('');
}

console.log(restoreString(s, indices));  // Output: "leetcode"